ppr-ui/tests/unit/RegistrationLengthTrustAmendment.spec.ts

```diff
@@ -0,0 +1,216 @@
+      showInvalid: false,
+      surrenderDate: '',
+      lienAmount: ''
+    })
+    await store.dispatch('setLengthTrust', {
+      valid: true,
+      trustIndenture: true,
+      lifeInfinite: false,
+      lifeYears: 5,
+      showInvalid: false,
+      surrenderDate: '',
+      lienAmount: ''
+    })
+    expect(wrapper.findComponent(RegistrationLengthTrustAmendment).exists()).toBe(true)
+    expect(wrapper.vm.showTrustIndenture).toBe(true)
+    expect(wrapper.vm.trustIndenture).toBe(true)
+    expect(wrapper.vm.showEditTrustIndenture).toBe(false)
+    expect(wrapper.vm.editInProgress).toBe(false)
+    expect(wrapper.vm.computedExpiryDateFormatted).toBeDefined()
+    expect(wrapper.find(undoButton).exists()).toBe(true)
+    wrapper.find(undoButton).trigger('click')
+    await Vue.nextTick()
+    expect(wrapper.find(amendButton).exists()).toBe(true)
+  })
+
+  it('renders with trust indenture false', async () => {
+    await store.dispatch('setOriginalLengthTrust', {
+      valid: true,
+      trustIndenture: false,
+      lifeInfinite: false,
+      lifeYears: 5,
+      showInvalid: false,
+      surrenderDate: '',
+      lienAmount: ''
+    })
+    await store.dispatch('setLengthTrust', {
+      valid: true,
+      trustIndenture: false,
+      lifeInfinite: false,
+      lifeYears: 5,
+      showInvalid: false,
+      surrenderDate: '',
+      lienAmount: ''
+    })
+    expect(wrapper.findComponent(RegistrationLengthTrustAmendment).exists()).toBe(true)
+    expect(wrapper.vm.showTrustIndenture).toBe(true)
+    expect(wrapper.vm.trustIndenture).toBe(false)
+    expect(wrapper.vm.showEditTrustIndenture).toBe(false)
+    expect(wrapper.vm.editInProgress).toBe(false)
+    expect(wrapper.vm.computedExpiryDateFormatted).toBeDefined()
+    expect(wrapper.find(amendButton).exists()).toBe(true)
+    wrapper.find(amendButton).trigger('click')
+    await Vue.nextTick()
+    expect(wrapper.findComponent(EditTrustIndenture).exists()).toBeTruthy()
+    expect(wrapper.findComponent(EditTrustIndenture).isVisible()).toBe(true)
+    expect(wrapper.vm.showEditTrustIndenture).toBe(true)
+    expect(wrapper.vm.editInProgress).toBe(true)
+    expect(wrapper.find(cancelButton).exists()).toBe(true)
+    wrapper.find(cancelButton).trigger('click')
+    await Vue.nextTick()
+    expect(wrapper.findComponent(EditTrustIndenture).exists()).toBeFalsy()
+    expect(wrapper.vm.showEditTrustIndenture).toBe(false)
+    expect(wrapper.vm.editInProgress).toBe(false)
+    wrapper.find(amendButton).trigger('click')
+    await Vue.nextTick()
+    expect(wrapper.vm.showEditTrustIndenture).toBe(true)
+    expect(wrapper.vm.editInProgress).toBe(true)
+    expect(wrapper.find(doneButton).exists()).toBe(true)
+    wrapper.find(doneButton).trigger('click')
+    await Vue.nextTick()
+    expect(wrapper.findComponent(EditTrustIndenture).exists()).toBeFalsy()
+    expect(wrapper.vm.showEditTrustIndenture).toBe(false)
+    expect(wrapper.vm.editInProgress).toBe(false)
+  })
+})
```

Comment:
Do we want a test for trust indenture change? to verify it changes in the store?

Reply:
I think the edit trust indenture component should have its own suite and it would cover this (I commented in my review)

Reply:
Added EditTrustIndenture tests which verify the store value..